File: server/Models/LandPlot.cs
namespace Server.Models;

/// <summary>
/// A portion of a <see cref="Farm"/> dedicated to one crop — e.g. "30 ha of potatoes" within a
/// larger farmland. What grows there is either a fruit the farm holds
/// (<see cref="TreeStockId"/>) or one of its stock goods (<see cref="StockId"/>), never both.
/// </summary>
public class LandPlot
{
    public int Id { get; set; }
    public int FarmId { get; set; }
    public decimal Area { get; set; }
    public string Crop { get; set; } = string.Empty;

    /// <summary>
    /// The <see cref="TreeStock"/> row growing here, picked from the fruits the farm holds. One
    /// plot per fruit within a piece of land, though other land may grow the same fruit. Null on
    /// plots recorded before a plot named its fruit, and on ones whose stock was deleted since.
    /// </summary>
    public int? TreeStockId { get; set; }

    /// <summary>
    /// The <see cref="Stock"/> row growing here, for plots of a field crop rather than fruit
    /// trees. Follows the same one-per-piece-of-land rule as <see cref="TreeStockId"/>, and is
    /// null whenever the plot names a fruit instead (or named neither, on older rows).
    /// </summary>
    public int? StockId { get; set; }
}

File: web/src/types/land-plot.ts
export type LandPlot = {
  id: number;
  farmId: number;
  area: number;
  crop: string;
  /** The fruit entry planted here, from the Fruits list. One plot per fruit on a piece of land,
   *  though other land may grow it too. Null on plots recorded before a plot named its fruit. */
  treeStockId: number | null;
  /** The stock good sown here, from the Stock list — the field-crop counterpart of
   *  `treeStockId`, under the same one-per-piece-of-land rule. A plot names one or the other,
   *  never both, so this is null whenever `treeStockId` is set. */
  stockId: number | null;
};

export type LandPlotInput = Omit<LandPlot, 'id'>;
